src/core/distributions/phylogenetics/tree/birthdeath/OccurrenceBirthDeathProcess.cpp

```diff
@@ -0,0 +1,543 @@
+ * \param[in]    inextinction              Extinction/death rate(s).
+ * \param[in]    inserialsampling          Serial sampling rate(s).
+ * \param[in]    intreatment               Probabilit(y|ies) of death upon sampling (treatment).
+ * \param[in]    inoccurrence              Occurrence sampling rate(s).
+ * \param[in]    ineventsampling           Sampling probability at present time.
+ * \param[in]    ht                        Rate interval change times of the piecewise constant process.
+ * \param[in]    cdt                       Condition of the process (survival/survival2).
+ * \param[in]    tn                        Taxa.
+ * \param[in]    uo                        If true the start age is the origin time otherwise the root age of the process.
+ * \param[in]    t                         Starting tree if we want to avoid simulating trees.
+ * \param[in]    n                         Maximum number of hidden lineages (algorithm accuracy).
+ * \param[in]    O                         Vector of occurrence ages.
+
+ */
+OccurrenceBirthDeathProcess::OccurrenceBirthDeathProcess(                                  const TypedDagNode<double> *sa,
+                                                                                           const DagNode *inspeciation,
+                                                                                           const DagNode *inextinction,
+                                                                                           const DagNode *inserialsampling,
+                                                                                           const DagNode *intreatment,
+                                                                                           const DagNode *inoccurrence,
+                                                                                           const DagNode *ineventsampling,
+                                                                                           const TypedDagNode< RbVector<double> > *ht,
+                                                                                           const std::string &cdt,
+                                                                                           const std::vector<Taxon> &tn,
+                                                                                           bool uo,
+                                                                                           Tree *t,
+                                                                                           const TypedDagNode<long> *n,
+                                                                                           const TypedDagNode< RevBayesCore::RbVector<double> > *O,
+                                                                                           bool mt,
+                                                                                           bool vb) : AbstractBirthDeathProcess( sa, cdt, tn, uo, t ),
```

Comment:
since you use the AbstractBirthDeathProcess, it might be helpful for MCMC analysis to allow providing a starting tree (see https://github.com/revbayes/revbayes/blob/development/src/core/distributions/phylogenetics/tree/birthdeath/EpisodicBirthDeathSamplingTreatmentProcess.cpp)